classes/submission/reviewAssignment/ReviewAssignmentDAO.inc.php

```diff
@@ -376,15 +375,21 @@ function insertObject($reviewAssignment) {
 		);
 
 		$reviewAssignment->setId($this->getInsertId());
-		return $reviewAssignment->getId();
+
+		// Update review stage status whenever a review assignment is changed
+		if ($result) {
+			$this->updateReviewRoundStatus($reviewAssignment);
+		}
+
+		return $reviewAssignment;
```

Comment:
I mean the `insertObject` functions. Usually they get the inserted ID, then call `setId` on the object. Since PHP5+ always treats objects as references, that propagates.